remapping_pipeline.nf

```diff
@@ -0,0 +1,350 @@
+
+    output:
+        path "old_ref_alleles.txt" into old_ref_alleles
+        path "variants_remapped.vcf" into variants_remapped
+
+    """
+    # Ensure that we will use the reverse_strand.py from this repo
+    ${baseDir}/reverse_strand.py -i reads_aligned.sorted.bam -p old_ref_alleles.txt -o variants_remapped.vcf -f $params.flankingseq -s $params.scorecutoff -d $params.diffcutoff 
+    """
+}
+
+/*
+ * Insert the reference base from the reference genome. 
+ */
+process insertReferenceAllele {
+
+    input:
+        path "variants_remapped.vcf" from variants_remapped
+        path "genome.fa" from params.newgenome
+        path "genome.fa.fai" from newgenome_fai
+
+    output:
+        path "var_pre_final.vcf" into var_pre_final
+
+    '''
+    # Add interval for variant position in bed format (required by getfasta)
+    # Reprints all the columns, adding an extra column before the pos column as the pos-1, as bedtools getfasta requires a 
+    # bed file
+    # Input: 
+    # [chr]	[pos]	[rsID]	[ALT]	[QUAL]	[FILT]	[INFO]
+    # Output:
+    # [chr]	[pos-1]	[pos]	[rsID]	[ALT]	[QUAL]	[FILT]	[INFO]
+    awk '{print $1"\\t"$2-1"\\t"$2"\\t"$3"\\t"$4"\\t"$5"\\t"$6"\\t"$7}' variants_remapped.vcf > variants_remapped.bed
+    
+    # Getfasta to get the REF genome alleles and then extract the genome alleles without the positions
+    bedtools getfasta -tab -fi genome.fa -bed variants_remapped.bed | awk '{print $2}'> genome_alleles.txt
+    
+    # Make all alleles caps (sometimes there are lower-case bases)
+    tr a-z A-Z < genome_alleles.txt > genome_alleles.fixed.txt
+    
+    # Paste all the information into the correct columns
+    cut -f1-3 variants_remapped.vcf > temp_first_3_columns.txt
+    cut -f4-7 variants_remapped.vcf > temp_last_4_columns.txt
+    paste temp_first_3_columns.txt genome_alleles.fixed.txt temp_last_4_columns.txt > var_pre_final.vcf
+    '''
+}
+
+
+/*
+ * 
+ */
+process buildHeader {
+
+    input:
+        path "var_pre_final.vcf" from var_pre_final
+        path "vcf_header.txt" from vcf_header
+
+    output:
+        path "final_header.txt" into final_header
+
+    """
+    # Create list of contigs/chromosomes to be added to the header
+    cut -f 1 var_pre_final.vcf | sort -u > contig_names.txt
+    while read CHR; do echo "##contig=<ID=\${CHR}>"; done < contig_names.txt > contigs.txt
+    # Add the reference assembly
+    echo "##reference=${params.newgenome}" >> contigs.txt
+    
+    # Copy everything that isn't #CHROM (the column titles), ##contig or ##reference from the old header to a temp
+    awk '(\$1 !~ /^##contig/ && \$1 !~ /^##reference/ && \$1 !~ /^#CHROM/) {print \$0}' vcf_header.txt > temp_header.txt
+
+    # Add the two headers together and add the column names
+    cat temp_header.txt contigs.txt > final_header.txt
+    echo -e "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO" >> final_header.txt
+    """
+}
+
+
+/*
+ * 
+ */
+process mergeHeaderAndContentAndFixRefAllele {
```

Comment:
I would actually recommend splitting this step into two: (1) merge header and content, (2) fix ref/alt alleles. Even though this second step is small, it is very distinct from the first one and is essentially a postprocessing step which should have a separate process.

Reply:
I think you're right.